src/h5web/providers/context.ts

```diff
@@ -2,19 +2,13 @@ import { createContext } from 'react';
 import type { HDF5Id, HDF5Value, HDF5Metadata } from './models';
 
 export abstract class ProviderAPI {
-  abstract getDomain: () => string;
-  abstract getMetadata: () => Promise<HDF5Metadata>;
-  abstract getValue: (id: HDF5Id) => Promise<HDF5Value>;
+  abstract domain: string;
+  abstract async getMetadata(): Promise<HDF5Metadata>;
+  abstract async getValue(id: HDF5Id): Promise<HDF5Value>;
 }
 
-function missing(): never {
-  throw new Error(
-    'Missing data provider context; please wrap `<H5Web />` with one of the available providers.'
-  );
-}
-
-export const ProviderContext = createContext<ProviderAPI>({
-  getDomain: missing as any, // eslint-disable-line @typescript-eslint/no-explicit-any
-  getMetadata: missing as any, // eslint-disable-line @typescript-eslint/no-explicit-any
-  getValue: missing as any, // eslint-disable-line @typescript-eslint/no-explicit-any
-});
+export const ProviderContext = createContext<{
+  domain: string;
+  metadata: HDF5Metadata;
+  getValue: ProviderAPI['getValue'];
+}>({} as any); // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
The app doesn't get rendered without metadata anymore, so providing an initial state in case users don't wrap the `App` in a provider seems less relevant.